File: src/main/java/cz/nitramek/network/UDPWorker.java
package cz.nitramek.network;

import java.nio.channels.DatagramChannel;


public abstract class UDPWorker implements ThreadedWorker {

    private DatagramChannel channel;

    @Override
    public void preWork() throws Exception {
        channel = DatagramChannel.open();
        preWork(channel);
    }


    @Override
    public void shutdown() throws Exception {
        channel.close();
    }


    @Override
    public void work() throws Exception {
        work(channel);
    }

    public void preWork(DatagramChannel channel) throws Exception {

    }

    public abstract void work(DatagramChannel datagramChannel) throws Exception;


}

File: src/main/java/cz/nitramek/messaging/network/UDPWorker.kt
package cz.nitramek.messaging.network

import java.nio.channels.DatagramChannel


abstract class UDPWorker : ThreadedWorker {

    private var channel: DatagramChannel? = null


    override fun preWork() {
        channel = DatagramChannel.open()
        preWork(channel!!)
    }


    override fun shutdown() {
        channel!!.close()
    }


    override fun work() {
        work(channel!!)
    }


    open fun preWork(channel: DatagramChannel) {
    }


    abstract fun work(datagramChannel: DatagramChannel)


}
